### PR TITLE
homework week1 browser Abdulrahman sadek

### PR DESCRIPTION
**Pull request by Abdulrahmansadek**

### CODE REVIEW

File: 2-Browsers/Week1/homework/ex3-hijackLogo.js [hunk 1]
```diff
@@ -8,7 +8,9 @@ Full description at: https://github.com/HackYourFuture/Homework/tree/main/2-Brow
    HackYourFuture logo instead.
 ------------------------------------------------------------------------------*/
 function hijackGoogleLogo() {
-  // TODO your code goes in here
+  const googleImage = document.getElementById('hplogo_new');
```
Comment (cgduncan7) [inline 2-Browsers/Week1/homework/ex3-hijackLogo.js:11]: well done 👏

File: 2-Browsers/Week1/homework/ex4-whatsTheTime/index.js [hunk 2]
```diff
@@ -7,8 +7,31 @@ Full description at: https://github.com/HackYourFuture/Homework/tree/main/2-Brow
...
+  } else if (minutes < 10) {
+    div.textContent = `${hours}  :  0${minutes} : ${seconds}`;
+  } else {
+    div.textContent = `${hours} :  ${minutes}  :  ${seconds}  `;
+  }
 }
+setInterval(addCurrentTime, 1000);
+window.addEventListener('load', addCurrentTime);
 
 // TODO execute `addCurrentTime` when the browser has completed loading the page
```
Comment (cgduncan7) [inline 2-Browsers/Week1/homework/ex4-whatsTheTime/index.js:37]: This `TODO` comment caused your tests to fail! Don't forget the clean-up 😄

File: 2-Browsers/Week1/homework/ex4-whatsTheTime/index.js [hunk 3]
```diff
@@ -7,8 +7,31 @@ Full description at: https://github.com/HackYourFuture/Homework/tree/main/2-Brow
...
+  const minutes = today.getMinutes();
+  const seconds = today.getSeconds();
+
+  if (seconds < 10) {
+    div.textContent = `${hours}  :  ${minutes} : 0${seconds}`;
+  } else if (minutes < 10) {
+    div.textContent = `${hours}  :  0${minutes} : ${seconds}`;
+  } else {
+    div.textContent = `${hours} :  ${minutes}  :  ${seconds}  `;
+  }
```
Comment (cgduncan7) [inline 2-Browsers/Week1/homework/ex4-whatsTheTime/index.js:32]: Hmm... I think there are some mistakes here. Let's take a look at an example.

Suppose the values of `hours`, `minutes`, and `seconds` are as defined below:
```javascript
hours = 1;
minutes = 1;
seconds = 1;
```

What will you write to the `div`? `seconds` is indeed less than 10 so the first `if-statement` will be true and we will have
```javascript
div.textContent = "1:1:01"
```
This doesn't look like a proper _HH:MM:SS_ time to me 😄

Try to rewrite this to solve this problem

File: 2-Browsers/Week1/homework/ex5-catWalk/index.js [hunk 4]
```diff
@@ -21,8 +21,46 @@ Full description at: https://github.com/HackYourFuture/Homework/tree/main/2-Brow
...
+          'http://www.anniemation.com/clip_art/images/cat-walk.gif';
+        walk = setInterval(walkBreakPoints, 50);
+      }, 5000);
+      //reset the left at the end
+    } else if (left >= windowWidth - 10 && left <= windowWidth) {
+      left = 0;
+
+      //normal situation
+    } else {
+      left += 10;
```
Comment (cgduncan7) [inline 2-Browsers/Week1/homework/ex5-catWalk/index.js:58]:
```suggestion
    } else if (left >= windowWidth - 10 && left <= windowWidth) {
      //reset the left at the end
      left = 0;
    } else {
      //normal situation
      left += 10;
```

Make sure the comments are in the correct block of code!

File: 2-Browsers/Week1/homework/ex5-catWalk/index.js [hunk 5]
```diff
@@ -21,8 +21,46 @@ Full description at: https://github.com/HackYourFuture/Homework/tree/main/2-Brow
...
+      walkingCat.style.left = left + 'px';
+    }
+  }
 }
 
 // TODO execute `catWalk` when the browser has completed loading the page
```
Comment (cgduncan7) [inline 2-Browsers/Week1/homework/ex5-catWalk/index.js:64]: Nasty `TODO` failing your tests again 👿

File: 2-Browsers/Week1/homework/ex5-catWalk/index.js [hunk 6]
```diff
@@ -21,8 +21,46 @@ Full description at: https://github.com/HackYourFuture/Homework/tree/main/2-Brow
...
+const walkingCat = document.querySelector('img');
+let left = 0;
+const width = 300;
+walkingCat.style.left = left + 'px';
+walkingCat.style.width = width + 'px';
+
 function catWalk() {
-  // TODO complete this function
+  //call an inner  function with setInterval
+  let walk = setInterval(walkBreakPoints, 50);
```
Comment (cgduncan7) [inline 2-Browsers/Week1/homework/ex5-catWalk/index.js:32]: 😹 👍

**Review by cgduncan7 — CHANGES_REQUESTED** [review]
Really nice work @Abdulrahmansadek 👏 

There are a few minor things I'd like to see changed though.

1. you are so close to all tests passing! Try to solve the minor issues
2. ex2 for-loop needs a revisit
3. ex4 time-string formatting needs a revisit

Once you do these it'll be 💯 👌

**Comment by Abdulrahmansadek** [comment]
hey, @cgduncan7 thanks a lot for reviewing my homework I'm gonna go through the comments. thanks a lot again.

**Comment by robvk** [comment]
@Abdulrahmansadek , please have a look at this. It is still open.

**Review by Abdulrahmansadek — COMMENTED** [review]
@robvk  this is the my modified homework and all the problems are solved .

**Pull request closed without merging**